pipelines/SpeciesTreeFromBusco/main.nf

```diff
@@ -217,14 +213,59 @@ process alignProt {
 
     input:
         val protFas
+        path cdnas
     output:
         path "alignments/prot_aln_*.fas", emit: prot_aln
+        path cdnas, emit: cdnas
    
     script:
     id = (protFas =~ /.*prot_(.*)\.fas$/)[0][1]
     """
     mkdir -p alignments
-    ${params.mafft_exe} --auto $protFas > alignments/prot_aln_${id}.fas
+    ${params.mafft_exe} --anysymbol --auto $protFas > alignments/prot_aln_${id}.fas
+    """
+}
+
+/**
+*@input path to protein alignment fasta
+*@output path to cDNA sequences fasta
+*@output path to codon alignment fasta
+*/
+process protAlnToCodon {
+    label 'rc_1Gb'
+
+    publishDir "${params.results_dir}/", pattern: "alignments/codon_aln_*.fas", mode: "copy",  overwrite: true
+
+    input:
+        path prot_aln
+        val cdna
+    output:
+        path "alignments/codon_aln_*.fas", emit: codon_aln
+
+    script:
+    id = (prot_aln =~ /.*prot_aln_(.*)\.fas$/)[0][1]
+    """
+    mkdir -p alignments
+
+    # Filter out from the cDNA sequences which did not pass protein
+    # level filtering:
+    ${params.seqkit_exe} -j 5 fx2tab -n $prot_aln > prot.ids
+    ${params.seqkit_exe} grep -j 5 -n -f prot.ids $cdna > filtered_cdna.fas
+    # Convert AA to codon alignment:
+    ${params.pal2nal_exe} $prot_aln filtered_cdna.fas -output fasta > alignments/codon_aln_${id}.fas.raw
+    if [ -s alignments/codon_aln_${id}.fas.raw ];
+    then
+        true;
+    else
+        echo "Codon alignment is empty!"
+        exit 1
+    fi
+    # Remove stop codons:
+    java -jar ${params.macse_jar} -prog exportAlignment \
+    -align alignments/codon_aln_${id}.fas.raw \
+    -codonForFinalStop --- \
+    -codonForInternalStop NNN \
+    -out_NT alignments/codon_aln_${id}.fas
```

Comment:
This is quite a wordy `script` in pluggability terms. I wonder if some of this can be split into more `processes` to meet the requirement of pluggability. The aim when writing new pipelines is ease of portability into another workflow manager. Can the filtering be moved to another separate process to the stop codon removal that flows into the stop codon removal?

Reply:
The stop codon removal indeed can be moved into a different process, so I will do this. The filtering step is too ad-hoc to be moved though.

Reply:
I only meant to split this process into two, so moving the stop codon removal into a different process achieves what I am asking :relieved:

Reply:
Cool, will do!